### PR TITLE
apps/system: Add tcpdump command

### PR DESCRIPTION
**Pull request by wengzhe**
## Summary
Add a simple tcpdump command to capture pcap file from network traffic.
The captured file can be recognized by wireshark.

Note: needs https://github.com/apache/nuttx/pull/8753 for full feature, otherwise we may only capture some of the packets.

```shell
> tcpdump                                                                                        
Usage:                                                                                                                              
  -i, --interface=interface      Capture device                                                                                     
  -w file                        Path to dump file                                                                                  
  -s, --snapshot-length=snaplen  Max dump length of each packet    
```
## Impact
Add a simple tcpdump command to capture pcap file from network traffic.

## Testing
Manually

### CODE REVIEW

File: system/tcpdump/tcpdump.c [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+ * Name: write_packet
+ ****************************************************************************/
+
+static int write_packet(int fd, uint32_t snaplen, uint32_t pkt_len,
+                        FAR const void *buf, FAR const struct timespec *ts)
+{
+  struct pcap_pkthdr_s hdr =
+    {
+      ts->tv_sec, ts->tv_nsec,
+      MIN(snaplen, pkt_len), pkt_len
```
Comment (pkarashchenko) [inline system/tcpdump/tcpdump.c:152]:
```suggestion
      ts->tv_sec,
      ts->tv_nsec,
      MIN(snaplen, pkt_len),
      pkt_len
```

File: system/tcpdump/tcpdump.c [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    {
+      perror("ERROR: failed to create packet socket");
+      return -errno;
+    }
+
+  /* Prepare sockaddr struct */
+
+  addr.sll_family = AF_PACKET;
+  addr.sll_ifindex = ifindex;
+  if (bind(sd, (const struct sockaddr *)&addr, sizeof(addr)) < 0)
```
Comment (pkarashchenko) [inline system/tcpdump/tcpdump.c:194]:
```suggestion
  if (bind(sd, (FAR const struct sockaddr *)&addr, sizeof(addr)) < 0)
```

File: system/tcpdump/tcpdump.c (hunk 1 above)
Comment (wengzhe) [reply system/tcpdump/tcpdump.c]: Done

File: system/tcpdump/tcpdump.c (hunk 2 above)
Comment (wengzhe) [reply system/tcpdump/tcpdump.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**